packages/status-js/src/client/account.ts

```diff
@@ -31,4 +46,34 @@ export class Account {
 
     return concatBytes(signature, new Uint8Array([recoverId]))
   }
+
+  updateMembership(community: Community): void {
+    const isMember = community.isMember('0x' + this.publicKey)
+
+    switch (this.membership) {
+      case 'none': {
+        community.requestToJoin()
+        this.membership = 'requested'
+        // fixme: this is a hack to make sure the UI updates when the membership status changes
```

Comment:
non-blocking: why are you calling a hack? Isn't it in the end the same thing as `callbacks.forEach()` which community, chats and activity center use?

Reply:
https://github.com/status-im/status-web/pull/315#discussion_r989002168